src/report.rs

```diff
@@ -26,7 +26,7 @@ pub struct Report {
 
 impl Report {
   /// Create a new instance.
-  pub fn new(method: Method, explanation: String) -> Self {
+  pub fn new<S: Into<String>>(name: S, version: S, method: Method, explanation: String) -> Self {
```

Comment:
This is… weird. What do you actually want to do? Use Cow, &str, or two type params?

Right now, you define _one_ type parameter that will end up being _one_ concrete type that can be converted to a String, and expect the first two function params to be of that one type.